lib/buffer.js

```diff
@@ -561,14 +557,6 @@ Buffer.prototype.write = function(string, offset, length, encoding) {
       encoding = length;
       length = undefined;
     }
-
-  // XXX legacy write(string, encoding, offset, length) - remove in v0.13
-  } else {
```

Comment:
Perhaps instead of flat out removing this, we should throw instead when `isFinite(offset)` is false?